c/array-fpi/brs1.c

```diff
@@ -0,0 +1,47 @@
+/*
+ * Benchmarks contributed by Divyesh Unadkat[1,2], Supratik Chakraborty[1], Ashutosh Gupta[1]
+ * [1] Indian Institute of Technology Bombay, Mumbai
+ * [2] TCS Innovation labs, Pune
+ *
+ */
+
+extern void __VERIFIER_error() __attribute__ ((__noreturn__));
+extern void __VERIFIER_assume(int);
+void __VERIFIER_assert(int cond) { if(!(cond)) { ERROR: __VERIFIER_error(); } }
+extern int __VERIFIER_nondet_int(void);
+
+int N;
+
+int main()
+{
+	N = __VERIFIER_nondet_int();
+	int i;
+	int sum[1];
+	int a[N];
```

Comment:
This allocates an array on the stack with an arbitrary size. First, `N` could be negative, which is clearly not ok.

Second, last year there was a discussion about unwanted large stack arrays and it was requested that they are either bounded or allocated on the heap (e.g., #561, #640, #787).

Reply:
@PhilippWendler I have modified the programs such that they won't create arrays of negative or zero size. This should help the tools that try to execute the programs for testing and verification. For the second point, the sizes are parametric and hence not necessarily large. Arrays with fixed size would defeat the purpose here as the tools must be able to verify the program irrespective of its size. I would like to keep them array manipulating rather than heap manipulating. A kind request to review the pull request. Many thanks.

Reply:
> @PhilippWendler I have modified the programs such that they won't create arrays of negative or zero size. This should help the tools that try to execute the programs for testing and verification. For the second point, the sizes are parametric and hence not necessarily large.

But they can be large and a verifier needs to handle that.

> Arrays with fixed size would defeat the purpose here as the tools must be able to verify the program irrespective of its size.

The array size could be bounded but still nondeterministic.

> I would like to keep them array manipulating rather than heap manipulating.

Then this needs to be discussed with those people who argued in favor of forbidding such programs in the previous years (cf. the issues I linked). For me both is fine.